### PR TITLE
Validate Chain ID

### LINKED ISSUE #245: Check RPC chain ID
## 📚 Context

The rollups node can fetch the chain ID from the Ethereum node RPC using the following method: https://ethereum-org-fork.netlify.app/developers/docs/apis/json-rpc#eth_chainid

Then, the node should compare the RPC chain ID with the configuration chain ID to make sure the node is executing in the correct chain.

### PR DESCRIPTION
**Pull request by torives**
Closes #245

### CODE REVIEW

File: cmd/cartesi-rollups-node/chainid.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			"chainId mismatch. Expected %v but Ethereum node returned %v",
+			chainId,
+			remoteChainId,
+		)
+	}
+	return nil
+}
+
+func getChainId(ethereumNodeAddr string) (uint, error) {
+	// create and send the request
```
Comment (gligneul) [inline cmd/cartesi-rollups-node/chainid.go:34]: Why not use go-ethereum to do the call?

**Review by gligneul — CHANGES_REQUESTED** [review]

File: cmd/cartesi-rollups-node/chainid.go (hunk 1 above)
Comment (torives) [reply cmd/cartesi-rollups-node/chainid.go]: There's no reason at all, I just forgot that was an option 🤦🏽‍♂️

File: cmd/cartesi-rollups-node/chainid.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	client, err := ethclient.Dial(ethereumNodeAddr)
+	if err != nil {
+		return 0, fmt.Errorf("Failed to create RPC client: %v", err)
+	}
+	chainId, err := client.ChainID(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("Failed to get chain id: %v", err)
+	}
+	return uint(chainId.Uint64()), nil
```
Comment (gligneul) [inline cmd/cartesi-rollups-node/chainid.go:45]: I don't feel comfortable with this cast. I think we should use uint64.

File: cmd/cartesi-rollups-node/chainid_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+type ValidateChainIdSuite struct {
+	suite.Suite
+}
+
+func TestValidateChainId(t *testing.T) {
+	suite.Run(t, new(ValidateChainIdSuite))
+}
+
+func (s *ValidateChainIdSuite) TestItFailsIfChainIdsDoNotMatch() {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (gligneul) [inline cmd/cartesi-rollups-node/chainid_test.go:25]: nice

**Review by gligneul — APPROVED** [review]

**Pull request merged**